src/components/Quotes.js

```diff
@@ -0,0 +1,39 @@
+import React, { useState, useEffect, useRef } from 'react';
+import './Quotes.css';
+
+const Quote = () => {
+  const [quotes, setQuotes] = useState('Loading...');
+  const textRef = useRef();
+  const colors = ['#ffff00', '#90ee90', '#ffa500', '#ff68ff', '#a9a9e7'];
+
+  const getQuotes = () => {
```

Comment:
- [ ] As suggested by the previous reviewer, you are supposed to have two more states apart from the quotes state and these are `loading` state and the `error` state. I can see that you tried to implement it by setting the initial value of the quotes to `loading...` but I think that is not the right way to do it. Basically these two other states are needed in order to keep of the fetching request and make sure we do handle different kinds of response cases and display something in the interface. I am going to share with you pieces of code that will help you get an idea of how to implement the suggested changes 👇 

```js
import { useEffect, useState } from "react"

function Quotes() {
  const [quotes, setQuotes] = useState([])
  const [hasError, setHasError] = useState(false)
  const [isLoading, setIsLoading] = useState(false)

  useEffect(() => {
//  FOR YOUR CASE THE BELOW CODES CAN STILL BE THERE IN THE `gateQuotes` function
    const fetchQuotes = async () => {
      setIsLoading(true) // SETTING THE LOADING STATE
      try {
        const res = await fetch(
          "https://api.api-ninjas.com/v1/quotes?category=happiness",
          {
            method: "GET",
            headers: {
              "X-API-Key": "API_KEY",
            },
          }
        )
        const json = await res.json()
        setQuotes(json)
      } catch (error) {
        setHasError(true) // SETTING ERROR IF THERE ARE
      }
      setIsLoading(false) // SETTING LOADING TO FALSE AS RESPONSE IS GOT ALREADY
    }
    fetchQuotes()
  }, [setQuotes, setIsLoading])

  if (hasError || Number(quotes.length) === 0)
    return <div>Something went wrong!</div> // This is an example of what could be returned

  if (isLoading) return <div>Loading...</div>

  return (
   // Below is an example of how you can render the quotes
    <ul>
      {quotes.map(({ author, quote }) => (
        <li key={author}>{quote}</li>
      ))}
    </ul>
  )
}

```
I hope that this will be useful, please make sure you add and use the above mentioned states 😃

Reply:
thanks a lot this really helps